src/matlab2tikz.m

```diff
@@ -407,9 +407,19 @@ function matlab2tikz(varargin)
 end
 % ==============================================================================
 function fid = fileOpenForWrite(m2t, filename)
-    encoding = switchMatOct({'native', m2t.cmdOpts.Results.encoding}, {});
```

Comment:
Oh and let me add: `isMATLAB()` and `isOctave()` is only available from within the test suites. (Greatness of those private functions...)